owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/http/methods/HttpBaseMethod.kt

```diff
@@ -13,24 +13,39 @@ import java.net.MalformedURLException
 import java.net.URL
 import java.util.concurrent.TimeUnit
 
-abstract class HttpBaseMethod constructor(clientWrapper: HttpClient, url: URL) {
-    var okHttpClient: OkHttpClient
+abstract class HttpBaseMethod constructor(url: URL) {
     var httpUrl: HttpUrl = url.toHttpUrlOrNull() ?: throw MalformedURLException()
     var request: Request
     abstract var response: Response
-
     var call: Call? = null
 
+    var followRedirects: Boolean = true
+    var retryOnConnectionFailure: Boolean = false
+    var connectionTimeoutVal: Long? = null
+    var connectionTimeoutUnit: TimeUnit? = null
+    var readTimeoutVal: Long? = null
+    var readTimeoutUnit: TimeUnit? = null
+
     init {
-        okHttpClient = clientWrapper.okHttpClient
         request = Request.Builder()
             .url(httpUrl)
             .build()
     }
 
     @Throws(Exception::class)
-    open fun execute(): Int {
-        return onExecute()
+    open fun execute(httpClient: HttpClient): Int {
+        val okHttpClient = httpClient.okHttpClient.newBuilder().apply {
+            retryOnConnectionFailure?.let { retryOnConnectionFailure(it) }
+            followRedirects?.let { followRedirects(it) }
```

Comment:
Now that they are not nullable, we can get rid of `?.let` here, can't we?
```suggestion
            retryOnConnectionFailure(retryOnConnectionFailure)
            followRedirects(followRedirects)
```